### PR TITLE
Features/xs 447 sensor malfunction new

### PR DESCRIPTION
**Pull request by nsztercsa**
Sensor malfunction flow

### CODE REVIEW

File: app/manage/pods/rules-form/conditional-form-directive/index.js [hunk 1]
```diff
@@ -1,6 +1,20 @@
 'use strict';
 var _ = require('lodash');
 
+var sensorOptions = [{id: 'temp', displayName: 'Temperature'},
```
Comment (tothandras) [inline app/manage/pods/rules-form/conditional-form-directive/index.js]: Do we need more hardcoded values? Isn't there another way?

File: app/manage/pods/rules-form/conditional-form-directive/template.tmpl [hunk 2]
```diff
@@ -10,21 +10,14 @@
...
       <div class="styled-select">
-        <select name='comparator' ng-model='rule.device'>
-          <option value='temp'>Temperature</option>
-          <option value='humidity'>Humidity</option>
-          <option value='no2'>NO2</option>
-          <option value='co'>CO</option>
-          <option value='dust'>Dust</option>
-          <option value='filter'>Filter</option>
-          <option value='fan'>Fan</option>
+        <select name='sensor' ng-model='rule.device' ng-model='rule.device' ng-options='c.id as c.displayName for c in comparatorOptions'>
```
Comment (tothandras) [inline app/manage/pods/rules-form/conditional-form-directive/template.tmpl]: bind once?

File: app/virtual-device/assets/css/purifier-controls.scss [hunk 3]
```diff
@@ -220,6 +221,62 @@
...
+
+  .purifier-device-toggle-btn:after {
+    background-color: rgb(12,232,14);
+  }
+}
+
+
+@keyframes blink{
+  0% {background-color: #4CB3E0;}
+  100% {background-color: rgb(255,72, 51);}
```
Comment (tothandras) [inline app/virtual-device/assets/css/purifier-controls.scss]: Why the mixed rgb with hex?

File: app/virtual-device/pods/common/factories/device-log-service-factory.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    function createDeviceLogMessage(inputData) {
+      var now = Date.now().toString();
+      var severity = inputData.severity || 'informational';
+      var code = '400';
+      if (severity === 'informational') {
+        code = '200';
+      }
+
+      return {
+        'sourceId': inputData.deviceId,
```
Comment (tothandras) [inline app/virtual-device/pods/common/factories/device-log-service-factory.js]: unnecessary qoutes

File: app/virtual-device/pods/common/index.js [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
 var randomInteger = require('./values/random-integer');
 var wiggle = require('./factories/wiggle');
+var deviceLogService = require('./factories/device-log-service-factory');
 var applicationConfig = require('../../../manage/pods/common/utils/application-config-constant');
 var loginProvider = require('../../../manage/pods/common/login-provider');
 var BlueprintClientProvider = require('../../../manage/pods/common/blueprint-client-provider');
+var states = require('./values/states');
 
-var commonModule = angular.module('common', []);
+var commonModule = angular.module('common', ['angular-uuid']);
```
Comment (tothandras) [inline app/virtual-device/pods/common/index.js]: this should be fine, most of the 3rd party modules return their name: `[require('angular-uuid')]`

File: app/virtual-device/pods/purifier/purifier-device/controller.js [hunk 6]
```diff
@@ -58,6 +68,86 @@ var purifierDeviceCtrl = [
...
+        templateId: logDevice.deviceTemplateId,
+        message: 'Sensor malfunction occured',
+        details: 'Sensor malfunction occured',
+        tags: ['malfunction']
+      };
+      deviceLogService.sendMalfunctionMessage(malfunctionData, deviceLogChannel);
+    }
+
+    function setDeviceState(state) {
+      $scope.safeApply(function() {
```
Comment (tothandras) [inline app/virtual-device/pods/purifier/purifier-device/controller.js]: `$applyAsync`?

File: app/manage/pods/rules-form/conditional-form-directive/index.js (hunk 1 above)
Comment (petercsala) [inline app/manage/pods/rules-form/conditional-form-directive/index.js]: The whole rule engine will be replaced soon.

That's why we have chosen the shortest and easiest way.

Comment (tothandras) [inline app/manage/pods/rules-form/conditional-form-directive/index.js]: okay, sure

**Comment by tothandras** [comment]
@nsztercsa Can you squash all of the commits into meaningful pieces? :) And please use the [commit message convention](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#type).

**Pull request merged**